File: src/MbUnit/Templates/VS2010/MbUnit.MvcWebApplicationTestProjectTemplate.CSharp/Controllers/HomeControllerTest.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using System.Web.Mvc;
using Gallio.Framework;
using MbUnit.Framework;
using $mvcprojectnamespace$;
using $mvcprojectnamespace$.Controllers;

namespace $safeprojectname$.Controllers
{
    /// <summary>
    /// Summary description for HomeControllerTest
    /// </summary>
    [TestFixture]
    public class HomeControllerTest
    {
        [Test]
        public void Index()
        {
            // Setup
            HomeController controller = new HomeController();

            // Execute
            ViewResult result = controller.Index() as ViewResult;

            // Verify
            ViewDataDictionary viewData = result.ViewData as ViewDataDictionary;

            Assert.AreEqual("Home Page", viewData["Title"]);
            Assert.AreEqual("Welcome to ASP.NET MVC!", viewData["Message"]);
        }

        [Test]
        public void About()
        {
            // Setup
            HomeController controller = new HomeController();

            // Execute
            ViewResult result = controller.About() as ViewResult;

            // Verify
            ViewDataDictionary viewData = result.ViewData as ViewDataDictionary;

            Assert.AreEqual("About Page", viewData["Title"]);
        }
    }
}

File: src/MbUnit/Templates/VS2008/MbUnit.MvcWebApplicationTestProjectTemplate.VisualBasic/Controllers/HomeControllerTest.vb
Imports System
Imports System.Text
Imports System.Collections.Generic
Imports System.Web.Mvc
Imports Gallio.Framework
Imports MbUnit.Framework
Imports $mvcprojectnamespace$
Imports $mvcprojectnamespace$.Controllers

<TestFixture()> _
Public Class HomeControllerTest
    <Test()> _
    Public Sub Index()
        ' Setup
        Dim controller As HomeController = New HomeController()

        ' Execute
        Dim result As ViewResult = controller.Index()

        ' Verify
        Dim viewData As ViewDataDictionary = result.ViewData
        
        Assert.AreEqual("Welcome to ASP.NET MVC!", viewData("Message"))
    End Sub

End Class
